File: scripts/debug_content.js
const fs = require('fs');

const ptData = JSON.parse(fs.readFileSync('data/fundamentos.json', 'utf8'));
const jaData = JSON.parse(fs.readFileSync('data/fundamentos_ja.json', 'utf8'));

const id = 'fundamentos_45';

const ptItem = ptData.find(i => i.id === id);
const jaItem = jaData.find(i => i.id === id);

if (!ptItem || !jaItem) {
    console.log("Item not found");
    process.exit(1);
}

const ptContent = ptItem.content;
let jaContent = jaItem.content;

// Retrieve regex used in modal.js
jaContent = jaContent.replace(/^\s*[（(].+?[）)]\s*/s, '').trim();

console.log("--- PT Content Analysis ---");
console.log("Raw Length:", ptContent.length);
console.log("Newlines:", ptContent.split('\n').length);
console.log("Double Newlines:", ptContent.split('\n\n').length);
console.log("Preview Lines:");
ptContent.split('\n').forEach((l, i) => console.log(`[${i}] ${l.substring(0, 20)}...`));

console.log("\n--- JA Content Analysis ---");
console.log("Raw Length:", jaContent.length);
console.log("Newlines:", jaContent.split('\n').length);
console.log("Double Newlines:", jaContent.split('\n\n').length);
console.log("Preview Lines:");
jaContent.split('\n').forEach((l, i) => console.log(`[${i}] ${l.substring(0, 20)}...`));

// Compare paragraph blocks (splitting by \n\n)
const ptBlocks = ptContent.split(/\n\n+/).filter(x => x.trim());
const jaBlocks = jaContent.split(/\n\n+/).filter(x => x.trim());

console.log("\n--- Block Comparison (split by \\n\\n) ---");
console.log(`PT Blocks: ${ptBlocks.length}`);
console.log(`JA Blocks: ${jaBlocks.length}`);

for (let i = 0; i < Math.max(ptBlocks.length, jaBlocks.length); i++) {
    console.log(`Block ${i}:`);
    console.log(`  PT: ${ptBlocks[i] ? 'Present' : 'MISSING'} (${ptBlocks[i] ? ptBlocks[i].length : 0} chars)`);
    console.log(`  JA: ${jaBlocks[i] ? 'Present' : 'MISSING'} (${jaBlocks[i] ? jaBlocks[i].length : 0} chars)`);
}

File: scripts/debug_content.py
import json
import re
import sys

# Load JSON
try:
    with open('data/fundamentos.json', 'r', encoding='utf-8') as f:
        pt_data = json.load(f)
    with open('data/fundamentos_ja.json', 'r', encoding='utf-8') as f:
        ja_data = json.load(f)
except FileNotFoundError:
    print("Files not found")
    sys.exit(1)

item_id = 'fundamentos_45'

# Find items
pt_item = next((i for i in pt_data if i['id'] == item_id), None)
ja_item = next((i for i in ja_data if i['id'] == item_id), None)

if not pt_item or not ja_item:
    print("Item not found")
    sys.exit(1)

pt_content = pt_item['content']
ja_content = ja_item['content']

# Apply JS regex logic: jaContent.replace(/^\s*[（(].+?[）)]\s*/s, '').trim()
# Note: Python regex for DOTALL is re.S
ja_content = re.sub(r'^\s*[（(].+?[）)]\s*', '', ja_content, flags=re.S).strip()

print("--- PT Content Analysis ---")
pt_lines = pt_content.split('\n')
print(f"Total Lines (split \\n): {len(pt_lines)}")
for i, line in enumerate(pt_lines):
    print(f"[{i}] {line[:20]}...")

print("\n--- JA Content Analysis ---")
ja_lines = ja_content.split('\n')
print(f"Total Lines (split \\n): {len(ja_lines)}")
for i, line in enumerate(ja_lines):
    print(f"[{i}] {line[:20]}...")

print("\n--- Block Analysis (split by \\n\\n) ---")
# Split by >= 2 newlines
pt_blocks = [b for b in re.split(r'\n\n+', pt_content) if b.strip()]
ja_blocks = [b for b in re.split(r'\n\n+', ja_content) if b.strip()]

print(f"PT Blocks: {len(pt_blocks)}")
print(f"JA Blocks: {len(ja_blocks)}")

max_blocks = max(len(pt_blocks), len(ja_blocks))
for i in range(max_blocks):
    pt_len = len(pt_blocks[i]) if i < len(pt_blocks) else 0
    ja_len = len(ja_blocks[i]) if i < len(ja_blocks) else 0
    print(f"Block {i}: PT={pt_len} chars, JA={ja_len} chars")


# Append logic to dump last blocks
print('
--- DETAILED BLOCK DUMP ---')
print('JA Block 8:')
print(ja_blocks[8])
print('
PT Block 8:')
print(pt_blocks[8])
print('
PT Block 9:')
print(pt_blocks[9])
print('
PT Block 10:')
print(pt_blocks[10])

